MirrorProvider/MirrorProvider.POSIX/POSIXNative.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Runtime.InteropServices;
+using System.Text;
+
+namespace MirrorProvider.POSIX
+{
+    public static class POSIXNative
+    {
+        public static string ReadLink(string path)
+        {
+            const ulong BufSize = 4096;
+            byte[] targetBuffer = new byte[BufSize];
+            long bytesRead = __ReadLink(path, targetBuffer, BufSize);
+            if (bytesRead < 0)
+            {
+                return null;
+            }
+
+            targetBuffer[bytesRead] = 0;
```

Comment:
What happens if we read exactly `BufSize` bytes? The [`man` page](http://man7.org/linux/man-pages/man2/readlink.2.html) appears to imply it will silently truncate, but can entirely fill the available bytes:

```
      readlink() places the contents of the symbolic link pathname in the
       buffer buf, which has size bufsiz.  readlink() does not append a null
       byte to buf.  It will (silently) truncate the contents (to a length
       of bufsiz characters), in case the buffer is too small to hold all of
       the contents.
```

Perhaps `byte[] targetBuffer = new byte[BufSize + 1];`?

Reply:
Or, should we throw an exception if we suspect the link was truncated (`bytesRead == BufSize`)?

Reply:
Indeed, I had shifted this code unchanged during this refactor, but I think you're right, and that this has the potential to crash, as it could return `BufSize` if the link target was exactly 4096 bytes.

I'm inclined to think throwing is cleaner, although right now the code expects a `null` for failure to `readlink` -- callers know how to deal with that.

We can use the three-argument variant of `Encoding.GetString` to avoid having to write the nul byte into the array ourselves, but we should still return null on `bytesRead == BufSize`, as it's ambiguous as to whether it was actually a 4096 byte target, or whether it was larger and truncated.

Changes made in 262f7c4c.